### PR TITLE
Modify the value saved in BridgeStorageProvider so that the second va…

### PR DESCRIPTION
**Pull request by pamgonzalez**
…lue is a simple value.

Change the information stored in BridgeStorageProvider. Now go on to save the hash of the derivation arguments with a 1-byte dummy value. This value is only used to save content together with the hash. It is only interesting to know if the hash of derivation arguments that is saved as key was calculated, it really does not matter what content is saved with this value.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 1]
```diff
@@ -625,22 +625,22 @@ private void saveDerivationArgumentsScriptHash() {
             return;
         }
 
-        fastBridgeDerivationArgumentsScriptHashToSave.forEach((derivationArgsHash, fastBridgeFedP2SH) ->
-                repository.addStorageBytes(contractAddress, getStorageKeyForDerivationP2SHByHash(derivationArgsHash), fastBridgeFedP2SH)
+        fastBridgeDerivationArgumentsScriptHashToSave.forEach((derivationArgsHash) ->
```
Comment (guidohernan93) [inline rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:628]: I think there are unnecessary parentheses surrounding `derivationArgHash`

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 2]
```diff
@@ -597,26 +600,23 @@ private void saveCoinbaseInformations() {
             safeSaveToRepository(getStorageKeyForCoinbaseInformation(blockHash), data, BridgeSerializationUtils::serializeCoinbaseInformation));
     }
 
-    public Optional<byte[]> getFastBridgeFederationScriptHash(Sha256Hash derivationArgsHash) {
+    public boolean isSavedFastBridgeFederationScriptHash(Sha256Hash derivationArgsHash) {
```
Comment (josedahlquist) [inline rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:603]:
```suggestion
    public boolean isFastBridgeFederationDerivationHashUsed(Sha256Hash derivationArgsHash) {
```

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 3]
```diff
@@ -597,26 +600,23 @@ private void saveCoinbaseInformations() {
...
         if (!activations.isActive(RSKIP176)) {
-            return Optional.empty();
+            return false;
         }
+        byte[] data = repository.getStorageBytes(contractAddress, getStorageKeyForDerivationARguments(derivationArgsHash));
```
Comment (josedahlquist) [inline rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:607]:
```suggestion
        byte[] data = repository.getStorageBytes(contractAddress, getStorageKeyForDerivationArguments(derivationArgsHash));
```

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 4]
```diff
@@ -597,26 +600,23 @@ private void saveCoinbaseInformations() {
...
-        if (fastBridgeFedP2SH == null ) {
-            return Optional.empty();
-        }
-
-        return Optional.of(fastBridgeFedP2SH);
+        return ((data != null) && (data.length == 1) && (data[0] == FAST_BRIDGE_FEDERATION_SCRIPT_HASH_TRUE_VALUE));
     }
 
-    public void setDerivationArgumentsScriptHash(Sha256Hash derivationArgsHash, byte[] fastBridgeFedP2SH) {
+    public void setDerivationArgumentsScriptHash(Sha256Hash derivationArgsHash) {
```
Comment (josedahlquist) [inline rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:612]:
```suggestion
    public void markFastBridgeFederationDerivationHashAsUsed(Sha256Hash derivationArgsHash) {
```

Comment (guidohernan93) [reply rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:612]: I think `setDerivationArgumentsScriptHash` is still there

**Review by josedahlquist — APPROVED** [review]

**Review by guidohernan93 — APPROVED** [review]

File: rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java [hunk 5]
```diff
@@ -111,9 +114,11 @@
 
     private Map<Sha256Hash, CoinbaseInformation> coinbaseInformationMap;
 
-    private Map<Sha256Hash, byte[]> fastBridgeDerivationArgumentsScriptHashToSave;
+    private Sha256Hash fastBridgeDerivationArgumentsScriptHashToSave = null;
+
+    byte[] fastBridgeScriptHash = new byte[]{};
```
Comment (guidohernan93) [inline rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java:119]: Should be private

**Pull request merged**